src/test/java/redis/clients/jedis/tests/ConnectionTest.java

```diff
@@ -75,4 +75,19 @@ public void sendCommand(ProtocolCommand cmd, byte[]... args) {
       assertEquals("ERR Protocol error: invalid multibulk length", jce.getMessage());
     }
   }
+
+  @Test
+  public void testReadProtocolWithBrokenState() {
+    client.connect();
+    try {
+      client.getOne();
+    } catch (JedisConnectionException ignored) {
+    }
```

Comment:
Will work on that, the reason for in the second commit I removed the assert because travic was throwing an exception that is different from the exception I've locally, although I don't care about what's the first exception, I care only about the second one, which is reading from a broken connection. But I'll add it anyway.